docs/tutorials/add-oauth2-provider.md

```diff
@@ -1,56 +1,178 @@
+        return $userId;
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserEmail(string $accessToken): string
+    {
+        // TODO: Fetch user from oauth API and select the user's email
+        $userEmail = "[FETCHED USER EMAIL]";
+        
+        return $userEmail;
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserName(string $accessToken): string
+    {
+        // TODO: Fetch user from oauth API and select the username
+        $username = "[FETCHED USERNAME]";
+        
+        return $username;
+    }
+}
+```
 
-Once a new class is created, you can start to implement your new provider's login flow. The best way to do this correctly is to have a look at another provider's implementation and try to follow the same standards.
+> If you copy this template, make sure to replace all placeholders wrapped like `[THIS]` and to implement everything marked as `TODO:`.
 
 Please mention in your documentation what resources or API docs you used to implement the provider's OAuth2 protocol.
 
-### Test Your Provider
+## 3. Test your provider
+
+After you finished adding your new provider to Appwrite, you should be able to see it in your Appwrite console. Navigate to 'Project > Users > Providers' and check your new provider's settings form.
 
-After you finished adding your new provider to Appwrite you should be able to see it in your Appwrite console. Navigate to 'Project > Users > Providers' and check your new provider's settings form.
+> To start Appwrite console from the source code, you can simply run `docker-compose up -d'.
 
-Add credentials and check both a successful and a failed login (where the user denies integration on provider page).
+Add credentials and check both a successful and a failed login (where the user denies integration on the provider page).
 
 You can test your OAuth2 provider by trying to login using the [OAuth2 method](https://appwrite.io/docs/client/account#accountCreateOAuth2Session) when integrating the Appwrite Web SDK in a demo app.
 
 Pass your new adapter name as the provider parameter. If login is successful, you will be redirected to your success URL parameter. Otherwise, you will be redirected to your failure URL.
 
-If everything goes well, submit a pull request and be ready to respond to any feedback which can arise during our code review.
+If everything goes well, raise a pull request and be ready to respond to any feedback which can arise during our code review.
+
+## 4. Raise a pull request
+
+First of all, commit the changes with the message `Added Slovak translations` and push it. This will publish a new branch to your forked version of Appwrite. If you visit it at `github.com/YOUR_USERNAME_appwrite`, you will see a new alert saying you are ready to submit a pull request. Follow the steps GitHub provides, and at the end, you will have your pull request submitted.
```

Comment:
You might need ot 
```suggestion
First of all, commit the changes with the message `Added XXX OAuth2 Provider` and push it. This will publish a new branch to your forked version of Appwrite. If you visit it at `github.com/YOUR_USERNAME/appwrite`, you will see a new alert saying you are ready to submit a pull request. Follow the steps GitHub provides, and at the end, you will have your pull request submitted.
```

Reply:
A similar URL mistake was in the translations tutorial, fixed that too.